### PR TITLE
Upgrade to latest DialogueKit

### LINKED ISSUE #171: Upgrade DialogueKit to v0.0.9.dev2
This implies to refactor ABUS to work with dialogue acts instead of single intents and to use `SlotValueAnnotation` where necessary.

### PR DESCRIPTION
**Pull request by NoB0**
Upgrade DialogueKit version to the most recent one (0.0.9.dev2)

## What's changed?

* Move the logic regarding the creation of response from ABUS to the interaction model
* Add `Agenda` class
* Use NLU with dialogue act extractor instead of intent classification and slot value annotation
* Update simulator to use an information need
* Update tests and dependent methods throughout the repo

Follow up issue #173 
Fixes #171

### CODE REVIEW

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 1]
```diff
@@ -1,19 +1,29 @@
 """Interaction model."""
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/interaction_model.py:1]: Could you please briefly describe here that this model also takes care of the agenda updates?

**Review by kbalog — APPROVED** [review]
Very nice work, LGTM.

I do have a question about the counts in the transition matrix, but that can be changed later if we decide so.

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 2]
```diff
@@ -243,93 +161,332 @@ def is_agent_intent_set_retrieval(self, agent_intent: Intent) -> bool:
...
+                .get(user_dialogue_act.intent.label, {})
+                .get("expected_agent_intents", [])
+            )
+
+        for agent_dialogue_act in agent_dialogue_acts:
+            if agent_dialogue_act.intent.label in expected_agent_intents:
+                return True
+
+        return False
+
```
[the remark this answers is not in the document]
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/interaction_model.py:203]: Good point, done.

File: usersimcrs/simulator/agenda_based/interaction_model.py (hunk 1 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/interaction_model.py:1]: Done.

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 3]
```diff
@@ -80,146 +96,48 @@ def _initialize_required_intents(self) -> None:
...
+        Dialogue acts in utt. 2: [INFORM(GENRE, action), INFORM(YEAR, 2024)]
 
-        Returns:
-            Intent distributions:
-                {user of agent intent: {next_user_intent: occurrence}}
+        The single intent transition matrix will be:
+        GREETING -> INFORM : 2
+        REQUEST -> INFORM : 4
+        The compound intent transition matrix will be:
+        {GREETING, REQUEST} -> INFORM : 2
```
[the remark this answers is not in the document]
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/interaction_model.py:113]: Good point, I updated the example to count on the utterance level.

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py [hunk 4]
```diff
@@ -181,66 +74,41 @@ def generate_response(
...
+        )
 
         # Test for the agent's stopping intent. Note that this would normally
         # handled by the dialogue connector. However, since intent annotations
         # for the agent's utterance are not available when the response is
         # received by the dialogue connector, an extra check is needed here.
-        if agent_intent == self._dialogue_connector._agent.stop_intent:
+        if self._dialogue_connector._agent.stop_intent in [
+            da.intent for da in agent_dialogue_acts
+        ]:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/agenda_based_simulator.py:89]: I like it, good suggestion.

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py [hunk 5]
```diff
@@ -21,40 +17,38 @@
...
         """Initializes the agenda-based simulated user.
 
         Args:
+            id: Simulator ID.
+            domain: Domain.
+            item_collection: Item collection.
             preference_model: Preference model.
             interaction_model: Interaction model.
-            nlu: NLU module performing intent classification and entity linking.
+            nlu: NLU module performing dialogue acts extraction.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/agenda_based_simulator.py:41]: Done.

**Pull request merged**